tests/unit/setup-unit-test-test.js

```diff
@@ -0,0 +1,75 @@
+import { setupUnitTest } from 'ember-mocha';
+import { describe, it, beforeEach, afterEach } from 'mocha';
+import { expect } from 'chai';
+import { pauseTest, resumeTest } from '@ember/test-helpers';
+import Service from '@ember/service';
+
+describe('setupUnitTest', function() {
+
+  describe('comtext setup', function() {
+
+    setupUnitTest();
+
+    afterEach(function() {
+      expect(this.owner, 'Context does not leak between tests').to.be.undefined;
+    });
+
+    it('sets up owner', function() {
+      expect(this.owner, 'Owner exists').to.exist;
+      this.owner.register('service:dummy', Service.extend({ foo: 'bar' }));
+
+      let subject = this.owner.lookup('service:dummy');
+      expect(subject.get('foo')).to.equal('bar');
+    });
+
+    it('has setters and getters', function() {
+      expect(this)
+        .to.respondTo('get')
+        .and.to.respondTo('getProperties')
+        .and.to.respondTo('set')
+        .and.to.respondTo('setProperties');
+
+      this.set('foo', 'bar');
+      expect(this.get('foo')).to.equal('bar');
+    });
+  });
+
+  describe('pauseTest/resumeTest', function() {
+
+    setupUnitTest();
+
+    it('can pause tests without timeouts', async function() {
+      this.timeout(100);
+
+      setTimeout(resumeTest, 200); // resume test after timeout
+      // make sure pauseTest does not wait forever, if resumeTest fails
+      let timer = setTimeout(() => expect(false, 'resumeTest() did not work').to.be.true, 300);
```

Comment:
did you check that this assertion works as expected if `resumeTest()` does indeed not work?

I think we should just `throw new Error('resumeTest() did not work')` instead of the always-failing assertion

Reply:
> did you check that this assertion works as expected if resumeTest() does indeed not work?

Yes, kind of. I removed the previous line (`setTimeout(resumeTest, 200);`), so not calling `resumeTest()`, and the assertion catched that.

> I think we should just throw new Error('resumeTest() did not work') instead of the always-failing assertion

Yeah, I tried that before, but somehow it was not working. Do you know, does chai just throw if the assertion fails, or is there some other magic going on? If not, I can try again...

Reply:
chai just throws exceptions (`AssertionError`), nothing else as far as I'm aware. It's a relatively simple system compared to QUnit.